app/domain/authentication/authn_k8s/k8s_object_lookup.rb

```diff
@@ -0,0 +1,110 @@
+# K8sObjectLookup is used to lookup Kubernetes object metadata using 
+# Kubernetes API.
+module Authentication
+  module AuthnK8s
+    module K8sObjectLookup
+      extend self
+
+      class K8sForbiddenError < RuntimeError
+      end
+
+      # Gets the client object to the /api v1 endpoint.
+      def kubectl_client
+        KubectlClient.client
+      end
+
+      # Locates the Pod with a given IP address.
+      #
+      # @return nil if no such Pod exists.
+      def find_pod_by_request_ip_in_namespace request_ip, namespace
```

Comment:
I'd lose `in_namespace` and also `find` (it returns an object, no side effects afict).  Same comment for other methods.  

When you're refactoring at some point ask me about how static utility classes are usually the sign of a hidden object waiting to be born, which will yield a more elegant solution.  It's a longer conversation better had in person.